app/controllers/document/RemoveDocumentController.scala

```diff
@@ -0,0 +1,85 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.document
+
+import controllers.{NavigatorOps, SettableOps, SettableOpsRunner}
+import controllers.actions._
+import forms.YesNoFormProvider
+import models.reference.Document
+import models.requests.SpecificDataRequestProvider1
+import models.{Index, LocalReferenceNumber, Mode}
+import pages.document.TypePage
+import pages.sections.DocumentDetailsSection
+import play.api.data.Form
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, Call, MessagesControllerComponents}
+import repositories.SessionRepository
+import uk.gov.hmrc.http.HttpVerbs.POST
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.document.RemoveDocumentView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class RemoveDocumentController @Inject() (
+  override val messagesApi: MessagesApi,
+  implicit val sessionRepository: SessionRepository,
+  actions: Actions,
+  getMandatoryPage: SpecificDataRequiredActionProvider,
+  formProvider: YesNoFormProvider,
+  val controllerComponents: MessagesControllerComponents,
+  view: RemoveDocumentView
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  private def form(documentType: Document): Form[Boolean] = formProvider("document.removeDocument", documentType)
+
+  private type Request = SpecificDataRequestProvider1[Document]#SpecificDataRequest[_]
+
+  private def documentType(implicit request: Request): Document = request.arg
+
+  def onPageLoad(lrn: LocalReferenceNumber, mode: Mode, documentIndex: Index): Action[AnyContent] = actions
+    .requireData(lrn)
+    .andThen(getMandatoryPage(TypePage(documentIndex))) {
+      implicit request =>
+        Ok(view(form(documentType), lrn, mode, documentIndex, documentType))
+    }
+
+  def onSubmit(lrn: LocalReferenceNumber, mode: Mode, documentIndex: Index): Action[AnyContent] = actions
+    .requireData(lrn)
+    .andThen(getMandatoryPage(TypePage(documentIndex)))
+    .async {
+      implicit request =>
+        lazy val redirect = Call(POST, "#")
+        form(documentType)
+          .bindFromRequest()
+          .fold(
+            formWithErrors => Future.successful(BadRequest(view(formWithErrors, lrn, mode, documentIndex, documentType))),
+            {
+              case true =>
+                DocumentDetailsSection(documentIndex)
```

Comment:
Should be `DocumentSection`. `DocumentDetailsSection` is nested inside `DocumentSection` to make cleanup easier when changing the document type.